### PR TITLE
feat: Remove user segment

### PR DESCRIPTION
**Pull request by AbhiPrasad**
ref https://github.com/getsentry/team-sdks/issues/44

Remove user segment and all user segment related logic.

https://go.dev/ref/mod#go-mod-file-module-deprecation

### CODE REVIEW

File: interfaces.go [hunk 1]
```diff
@@ -107,13 +107,17 @@ type Attachment struct {
...
+	ID        string `json:"id,omitempty"`
+	Email     string `json:"email,omitempty"`
+	IPAddress string `json:"ip_address,omitempty"`
+	Username  string `json:"username,omitempty"`
+	Name      string `json:"name,omitempty"`
+	// Segment is a string representing the user segment that the user belongs to.
+	//
+	// Deprecated: Segment is deprecated and will be removed in the next major
+	// of the Sentry Go SDK.
+	Segment string            `json:"segment,omitempty"`
```
Comment (cleptric) [inline interfaces.go:119]: Feel free to remove it right now, the next release contains some breaking changes as well.

**Review by cleptric — APPROVED** [review]

**Comment by AbhiPrasad** [comment]
@cleptric I went ahead and completely removed it in d66770f6e46f90f3a223896dbbd2e8b8d44e123e

**Comment by ribice** [comment]
@AbhiPrasad Please update Changelog as well with this change.

**Comment by AbhiPrasad** [comment]
@ribice done with 8b89a4ed4e2cf08f8388212fa34edb4676343bf9

**Comment by cleptric** [comment]
Thanks @AbhiPrasad

**Pull request merged**